### PR TITLE
[FEATURE] Migrate Querybuilder->execute()

### PR DESCRIPTION
**Pull request by helsner**
Resolves: #2938

### CODE REVIEW

File: tests/Rector/v12/v0/typo3/MigrateQueryBuilderExecuteRector/Fixture/fixture.php.inc [hunk 1]
```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Ssch\TYPO3Rector\Tests\Rector\v12\v0\typo3\MigrateQueryBuilderExecuteRector\Fixture;
+use TYPO3\CMS\Core\Database\ConnectionPool;
```
Comment (simonschaufi) [inline tests/Rector/v12/v0/typo3/MigrateQueryBuilderExecuteRector/Fixture/fixture.php.inc:4]: empty line after namespace

File: tests/Rector/v12/v0/typo3/MigrateQueryBuilderExecuteRector/Fixture/fixture.php.inc [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    ->from($table)
+    ->execute()
+    ->fetchAllAssociative();
+
+$count = $queryBuilder
+    ->count('*')
+    ->from($table)
+    ->execute()
+    ->fetchAllAssociative();
+
```
Comment (simonschaufi) [inline tests/Rector/v12/v0/typo3/MigrateQueryBuilderExecuteRector/Fixture/fixture.php.inc:42]: only one empty line

File: tests/Rector/v12/v0/typo3/MigrateQueryBuilderExecuteRector/Fixture/fixture.php.inc [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+$insertedRows = $queryBuilder
+    ->insert($table)
+    ->values(['foo', 'bar'])
+    ->execute();
+
+$deletedRows = $queryBuilder
+    ->delete($table)
+    ->execute();
+
```
Comment (simonschaufi) [inline tests/Rector/v12/v0/typo3/MigrateQueryBuilderExecuteRector/Fixture/fixture.php.inc:57]: only one empty line

File: composer.json [hunk 4]
```diff
@@ -72,7 +72,9 @@
         "sort-packages": true,
         "platform-check": false,
         "allow-plugins": {
-            "phpstan/extension-installer": true
+            "phpstan/extension-installer": true,
+            "cweagans/composer-patches": true,
```
Comment (simonschaufi) [inline composer.json:76]: why are they added? We don't use them any more.

**Review by simonschaufi — CHANGES_REQUESTED** [review]

**Comment by helsner** [comment]
We got an issue with the last test i added.
Sometimes the query is build with conditions etc. in between, so the chain does not happen directly, but is split into multiple pieces, with only the execute() at the end.
Either we leave this, then i'll adapt the test or we can fix it.
At least i prevented it from being falsely refactored

**Comment by helsner** [comment]
I will take care of the phpstan stuff when the actual rule is done.

File: tests/Rector/v12/v0/typo3/MigrateQueryBuilderExecuteRector/Fixture/fixture.php.inc [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+/** @var QueryBuilder $queryBuilderSplitSecond */
+$queryBuilderSplitSecond = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable($table);
+
+$testSecond = $queryBuilderSplitSecond
+    ->select('*')
+    ->from($table);
+
+$result = $testSecond
```
[the remark this answers is not in the document]
Comment (helsner) [reply tests/Rector/v12/v0/typo3/MigrateQueryBuilderExecuteRector/Fixture/fixture.php.inc:75]: i talked to Tomas, this is pretty pretty hard and would only work with assumptions.
I will provide a hint functionality as we can identify the cases but not migrate them.

**Pull request merged**